### PR TITLE
[Fluent 2 iOS] Bottom Commanding Controller

### PR DESCRIPTION
**Pull request by edjamesmsft**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Update the Bottom Commanding Controller to Fluent 2 designs. This builds off of Bottom Sheet.

### Verification

Built and ran the simulator, using the BottomCommandingControllerDemo in light and dark mode on multiple screens and different orientations. Double checked color values with colour value analyser to ensure colors match figma.

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Simulator Screen Shot - iPhone 13 Pro Max - 2022-08-24 at 15 13 04 | Simulator Screen Shot - iPhone 13 Pro Max - 2022-08-24 at 15 11 21 |
Simulator Screen Shot - iPhone 13 Pro Max - 2022-08-24 at 15 13 07 | Simulator Screen Shot - iPhone 13 Pro Max - 2022-08-24 at 15 11 25 |
Simulator Screen Shot - iPhone 13 Pro Max - 2022-08-24 at 15 13 34 | Simulator Screen Shot - iPhone 13 Pro Max - 2022-08-24 at 15 11 55 |
Simulator Screen Shot - iPhone 13 Pro Max - 2022-08-24 at 15 13 38 | Simulator Screen Shot - iPhone 13 Pro Max - 2022-09-02 at 13 40 55 |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [x] Different resolutions (1x, 2x, 3x)
- [] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1187)

### CODE REVIEW

**Review by mischreiber — COMMENTED** [review]
The dark demo looks very strange with the black section header bar. Is that just a quirk of our demo controller or indicative of a design oversight?

**Comment by edjamesmsft** [comment]
> The dark demo looks very strange with the black section header bar. Is that just a quirk of our demo controller or indicative of a design oversight?

Good call - looks like some odd behavior in how TableViewHeaderFooterView is consumed. Will dig into this more and see what needs to be done to fix it.

File: ios/FluentUI/Bottom Commanding/BottomCommandingController.swift [hunk 1]
```diff
@@ -854,21 +855,23 @@ open class BottomCommandingController: UIViewController {
...
         static let heroButtonWidth: CGFloat = 96
         static let heroButtonLabelMaxWidth: CGFloat = 72
         static let heroButtonMaxTitleLines: Int = 2
 
-        static let tableViewIconTintColor: UIColor = Colors.textSecondary
-        static let tableViewBackgroundColor: UIColor = Colors.NavigationBar.background
-
         struct BottomBar {
-            static let height: CGFloat = 80
+            static let height: CGFloat = 74
```
Comment (harrieshin) [inline ios/FluentUI/Bottom Commanding/BottomCommandingController.swift:871]: this is bit surprising to me. usually things snap to 4px grid.

File: ios/FluentUI/Bottom Commanding/BottomCommandingController.swift [hunk 2]
```diff
@@ -415,12 +415,13 @@ open class BottomCommandingController: UIViewController {
     private func makeBottomBarByEmbedding(contentView: UIView) -> UIView {
         let bottomBarView = UIView()
         let bottomBarLayer = bottomBarView.layer
-        bottomBarLayer.shadowColor = Constants.BottomBar.Shadow.color
-        bottomBarLayer.shadowOpacity = Constants.BottomBar.Shadow.opacity
```
Comment (laminesm) [inline ios/FluentUI/Bottom Commanding/BottomCommandingController.swift:419]: Don't we need to set the shadow opacity to 1?

**Review by huwilkes — APPROVED** [review]

**Pull request merged**